### PR TITLE
Feature/rework codesigning for mac

### PR DESCRIPTION
**Pull request by kerner1000**

### CODE REVIEW

**Comment by kerner1000** [comment]
Dear Sirs,
I moved the code signing to a dedicated library, which is already available on Maven Central. Please have a look and merge.
Next feature is notarisation, which depends on this changes.
How is release of 1.7.0 going?

**Comment by fvarrui** [comment]
> How is release of 1.7.0 going?

I'm afraid the release of 1.7.0 will be delayed, but we can release 1.6.8 with what's currently there, what do you think?

**Comment by kerner1000** [comment]
Sure, why not?
I think there speaks nothing against it if you make a release from the current state of devel.
I just brought another project to maven central, so if you need input on releasing, let me know, we can have a chat :)

**Comment by kerner1000** [comment]
If it breaks backwards compatibility the new release should be 1.7.0. but the current 1.7.0 can be as well 1.8.0 or 1.7.1, it really doesn't matter, does it?

**Comment by fvarrui** [comment]
I'm trying to follow (more or less) semantic versioning, so given a version number X.Y.Z, increment the:

- X when we make changes that break the backward compatibility
- Y when we add new features in a backwards compatible manner
- Z when we add patches
- Additional label "SNAPSHOT" when it's a pre-release version.

Actually, the next version to release should be 1.7.0, since new features have been added, but in I planned to add support for Java Modules in 1.7.0, and I haven't been able to complete it yet 😞

**Comment by kerner1000** [comment]
Understood, that makes sense! But why needs the Java Modules be version 1.7.0? Can you not change it to 1.8.0?
Anyways, what needs to be done to enable Modules? I am working on it right now, since a non-modular layout seems to make problems with JavaFX.
So far, I found a way to do it with minimal changes.
But I figured it would be maybe useful to do the packaging with jpackage, what do you think about this?
Maybe you could enable "discussions" on this repo so we have place for general discussions?

**Comment by fvarrui** [comment]
> But why needs the Java Modules be version 1.7.0? Can you not change it to 1.8.0? 

Just because I'm a stubborn 😄 ... there's no other reason

> Anyways, what needs to be done to enable Modules?

I think it would help improve custom JRE generation among other things.

> I am working on it right now, since a non-modular layout seems to make problems with JavaFX. So far, I found a way to do it with minimal changes. 

I've managed to run a modular Java app on Windows (just passing some VM arguments to Launch4j EXE) with success and  I think it should be easy on Linux and MacOS, but not tried yet.

> But I figured it would be maybe useful to do the packaging with jpackage, what do you think about this? Maybe you could enable "discussions" on this repo so we have place for general discussions?

Yes, this was proposed some time ago, but it remains to be seen if it covers all the functionality that JavaPackager offers. It's a great idea to open a discussion about it!

**Comment by fvarrui** [comment]
[jpackage discussion](https://github.com/fvarrui/JavaPackager/discussions/292) 😅

File: src/main/java/io/github/fvarrui/javapackager/packagers/MacPackager.java [hunk 1]
```diff
@@ -193,15 +194,16 @@ private File preparePrecompiledStartupStub() throws Exception {
...
 	}
 
-	private void codesign(String developerId, File entitlements, File appFile) throws Exception {
+	private void codesign(String developerId, File entitlements, File appFile, File executable) throws Exception {
 
-		prepareEntitlementFile(entitlements);
+		entitlements = prepareEntitlementFile(entitlements);
 
-		manualDeepSign(appFile, developerId, entitlements);
+		Signer signer = new Signer(developerId, appFile.toPath(), executable.toPath(), entitlements.toPath(), entitlements.toPath());
```
Comment (fvarrui) [inline src/main/java/io/github/fvarrui/javapackager/packagers/MacPackager.java:201]: So ... no need to call the `codesign` command anymore?

**Pull request closed without merging**